compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrDeclarationStorage.kt

```diff
@@ -262,6 +263,9 @@ class Fir2IrDeclarationStorage(
                 irDeclaration.parent = irClass
             }
         }
+        if (regularClass.isFromLibrary) {
+            annotationGenerator?.generate(irClass, regularClass)
+        }
```

Comment:
Should have been more clear in my earlier comment: I suggested a util function for the three-line pattern

Reply:
Aha, make sense. Thanks for the suggestion.